macros.d/macros.initrd

```diff
@@ -3,9 +3,23 @@
 # regenerated
 # See also fate#313506
 
+# mkdir is part of the coreutils package. When the post scriptlet of some
+# package using regenerate_initrd_post is run, the mkdir command may be
+# unavailable. This is non-fatal, because in all cases that matter for initrd
+# generation, coreutils will be part of the same transaction and will trigger
+# an initrd rebuild later. See boo#1217775.
+# 
+# Anyway, packages can use the regenerate_initrd_requires macro below to make
+# sure mkdir is available in their post scriptlet. Use if this macro is not
+# mandatory.
+
+%regenerate_initrd_requires \
+Requires(post): /usr/bin/mkdir \
+%nil
+
 %regenerate_initrd_post \
-	mkdir -p /run/regenerate-initrd/ \
-	touch /run/regenerate-initrd/all \
+        ! command -v mkdir >/dev/null || mkdir -p /run/regenerate-initrd/; \
+	[ ! -d /run/regenerate-initrd ] || > /run/regenerate-initrd/all; \
```

Comment:
`{ /usr/bin/mkdir -p /run/regenerate-initrd && > /run/regenerate-initrd/all;:;} 2>/dev/null`

Reply:
I tend to avoid `&&` because it makes pipelines fail. That's a matter of taste of course.

Also, your could would fail if `mkdir` didn't exist but `/run/regenerate-initrd` did (I am not sure if this happens in practice today, but it could e.g. if the installer created that directory before running the rpm transaction).

Reply:
Right, needs to be `{ /usr/bin/mkdir -p /run/regenerate-initrd && > /run/regenerate-initrd/all; } 2>/dev/null || :`

mkdir -p doesn't fail if the directory exists